src/apps/appstore/app.js

```diff
@@ -102,12 +189,44 @@ define(function(require) {
 							next(null, data.data);
 						}
 					});
+				},
+				marketplace: function(next) {
+					// use this fake to test ui without calling API
+					//
+					// let fake = {
+					// 	cluster_id: 'lolololololol',
+					// 	enabled: false,
+					// 	app_exchange_url: 'http://localhost:8080/',
+					// 	marketplace_url: 'http://localhost:3030',
+					// 	is_linked: false,
+					// 	_read_only: {}
+					// };
+					// next(null, fake);
+
+					if (!shouldShowMarket) {
+						return next();
+					}
+					monster.request({
+						resource: 'marketplace.get',
+						success: function(response) {
+							next(null, response.data);
+						},
+						error: function() {
+							next();
```

Comment:
Still unsure what will happen here.

Is the `/marketplace` endpoint expected to always be available, or will that be conditioned by a system flag (like `/external_numbers` for example)?

If the former, then this is fine, otherwise, maybe we can hide the `Marketplace Connector` button if this errors out?